src/librustc_span/def_id.rs

```diff
@@ -152,6 +152,19 @@ pub struct DefId {
     pub index: DefIndex,
 }
 
+/// Represents a potentially invalid `DefId`. This is used during incremental
+/// compilation to represent a `DefId` from the *previous* compilation session,
+/// which may no longer be valid. This is used to help map a `DefPathHash`
+/// to a `DefId` in the current compilation session.
+#[derive(Encodable, Decodable, Copy, Clone, Debug)]
+pub struct RawDefId {
```

Comment:
This structure is only used in `on_disk_cache.rs` so it can be moved there and privatized.